modules/research-framework/research-service/src/main/java/org/apache/airavata/research/service/handlers/ResourceHandler.java

```diff
@@ -177,6 +180,23 @@ public Resource getResourceById(String id) {
         return opResource.get();
     }
 
+    public boolean deleteResourceById(String id) {
+        Optional<Resource> opResource = resourceRepository.findById(id);
+
+        if (opResource.isEmpty()) {
+            throw new RuntimeException("Resource not found: " + id);
+        }
+
+        String userEmail = UserContext.userId();
+        System.out.println("Received request to delete with user: " + userEmail);
+        if (!opResource.get().getAuthors().contains(userEmail.toLowerCase())) {
```

Comment:
@ganning127 shall we add a TODO comment here saying that only the owner should be able to delete a resource?
Once we integrate roles and sharing we should restrict this feature